### PR TITLE
Restricting hosts where Python PreCommit runs due to environmental issues.

### PR DESCRIPTION
**Pull request by pabloem**

### CODE REVIEW

**Comment by pabloem** [comment]
Host choice is due to having seen passing tests earlier today:
beam1 - https://builds.apache.org/job/beam_PreCommit_Python_GradleBuild/496/consoleFull
beam6 - https://builds.apache.org/job/beam_PreCommit_Python_GradleBuild/487/consoleFull
beam7 Has been recently reimaged, so trying that one as well.

I see beam2 also had [a successful run](https://builds.apache.org/job/beam_PreCommit_Python_GradleBuild/499/consoleFull). I'll add it if tests pass.

**Comment by pabloem** [comment]
Hmm still failing. Something wrong with proto generation. Observing it in my machine as well.

**Comment by pabloem** [comment]
No luck. Still not good information on why it's failing.

**Comment by pabloem** [comment]
Failure is different. I think that's a good thing. Working on making progress.

File: sdks/python/gen_protos.py [hunk 1]
```diff
@@ -146,7 +146,8 @@ def _install_grpcio_tools_and_generate_proto_files():
         'Installing grpcio-tools took %0.2f seconds.', time.time() - start)
   finally:
     sys.stderr.flush()
-    shutil.rmtree(build_path)
+    if os.path.exists(build_path):
```
[the remark this answers is not in the document]
Comment (pabloem) [reply sdks/python/gen_protos.py]: Ah I'm sorry. Good memory. We had not merged that fix, but I'll make the change now.

**Comment by pabloem** [comment]
I've checked with Jason. I asked why Python tests pass/fail on some machines:

> The reasons for that are as many as the stars in the sky
> Different package versions, transient network issues, permanent network issues, Jenkins instability, etc.
> For Python, historically there were some machines with up-to-date pip and virtualenv and some without, and we restricted execution of Python tests to the ones with

It doesn't seem that old Maven-based precommit had any sort of restrictions on hosts, so I'm not sure what's going on in this case: https://github.com/apache/beam/blob/a25157bc6a9203362f6dd23acc8b7e1f30daad8e/.test-infra/jenkins/job_beam_PreCommit_Python_MavenInstall.groovy

Somehow it must be that the Gradle build and the Maven build had different assumptions?

**Comment by pabloem** [comment]
Run Seed Job

**Comment by pabloem** [comment]
Retest this please

**Comment by robertwb** [comment]
The maven-based precommit didn't use virtualenv (it modified and used the users install of Python). This should be more hermetic, assuming a recent virtualenv and pip.

**Pull request merged**